ur_driver/src/ur_driver/driver.py

```diff
@@ -464,9 +464,14 @@ def handle(self):
             print "Connection closed (command):", ex
             setConnectedRobot(None)
 
+    # Send a message to the robot.
+    def __send_message(self, params):
+        # Acquire a lock first to prevent race conditions.
+        with self.__socket_lock:
+            self.request.send(struct.pack("!%ii" % len(params), *params))
+
```

Comment:
I prefer to keep ```buf = struct.pack(...)``` in a separate line. There's no need to do this inside the lock, and I try to avoid putting multiple statements on one line. Just my two cents :)